### ISSUE TITLE
Swiching masters can cancel page breaking

### ISSUE THREAD

**Issue by Omikhleia** [sile-typesetter/sile#1580]
SILE 0.14.3

(Issue from discussion  #1579)

Using `\swich-master` in a SILE document does set the relevant master, **but** page breaking then no longer works when the main content frame is filled.

**Script**
```
\begin[papersize=letter,class=plain]{document}
\use[module=packages.masters]
\use[module=packages.lorem]
\define-master-template[id=mymaster,first-content-frame=content]{
  \frame[id=content,left=1in, right=100%pw-1in, top=1in, bottom=100%ph-1in]
  \frame[id=folio,left=1in,right=100%pw-1in,top=bottom(content)+0.2in,height=0.5in]
}
\switch-master[id=mymaster]
\lorem[words=2000]

\end{document}
```

**Observed result**
The frames are set as specified in the master template, but new pages fail to be generated when the content fills the current frame. The content frame overflows on the current page.

**Workaround**
Enforcing the master on the current page, _after_ the master is set, avoids the issue, i.e.
```
...
\switch-master[id=mymaster]
\switch-master-one-page[id=mymaster]
```

**Discussion**

- Page breaking is disabled because the current frame's `isMainContentFrame()` returns false. Adding some debug code here, it turns out the current frame has the correct definition but is not found in the current page templates. (There is one with the appropriate definition, but it's not the same object).

- FWIW, this might be the cause:

  https://github.com/sile-typesetter/sile/blob/751eeaf4376997b97c084f038e18edc45ea5a426/packages/masters/init.lua#L64

  There is no explanation there, but the deep copy is what makes the frames differ. Removing it also avoids the issue... It also explains why add `\switch-master-one-page` works: the latter does set the master without deep-copying it.

  https://github.com/sile-typesetter/sile/blob/751eeaf4376997b97c084f038e18edc45ea5a426/packages/masters/init.lua#L48

  I've no idea however why the deep copy was needed in the first place and whether removing it is an appropriate fix or would break something else.

**Comment by alerque** [comment]
Hm. I just fixed #1586 via #1589 but not I'm wondering if this bug isn't *also* part of the problem.

**Comment by alerque** [comment]
The `:isMainCantentFrame()` method relies on an `==` comparison between frames, which will return false for an otherwise identical object that has been deep-copies because the ids of tables inside it are not the same. I'll look into why the deepcopy is being done but at first blush I'm guessing it is safe to drop now that we have some better separation between packages and classes.
